src/org/jitsi/impl/neomedia/rtp/translator/SimulcastController.java

```diff
@@ -157,102 +134,118 @@ public long getTargetSSRC()
+        }
+
+        // if the TL0 of the forwarded stream is suspended, we MUST downscale.
+        boolean currentTL0IsActive = false;
+        int currentTL0Idx = bitstreamController.getCurrentIndex();
+        if (currentTL0Idx > -1)
+        {
+            currentTL0Idx
+                = sourceEncodings[currentTL0Idx].getBaseLayer().getIndex();
+
+            currentTL0IsActive = sourceEncodings[currentTL0Idx].isActive();
+        }
 
-            currentRTPEncodingIsActive = currentRTPEncodingDesc.isActive();
+        int targetTL0Idx = bitstreamController.getTargetIndex();
+        if (targetTL0Idx > -1)
+        {
+            targetTL0Idx
+                = sourceEncodings[targetTL0Idx].getBaseLayer().getIndex();
         }
 
-        if (transformState.currentIdx == targetIdx && currentRTPEncodingIsActive)
+        if (currentTL0Idx == targetTL0Idx && currentTL0IsActive)
         {
-            boolean accept = sourceSSRC == transformState.currentSSRC;
+            // An intra-codec/simulcast switch pending is NOT pending.
+            long sourceSSRC = sourceFrameDesc.getRTPEncoding().getPrimarySSRC();
+            boolean accept = sourceSSRC == bitstreamController.getTL0SSRC();
 
-            if (accept)
+            if (!accept)
             {
-                onAccept(buf, off, len);
+                return false;
             }
 
-            return accept;
+            // Give the bitstream filter a chance to drop the packet.
+            return bitstreamController.accept(sourceFrameDesc, buf, off, len);
         }
 
         // An intra-codec/simulcast switch pending.
 
-        FrameDesc sourceFrameDesc
-            = sourceTrack.findFrameDesc(buf, off, len);
-
-        int sourceIdx = sourceFrameDesc.getRTPEncoding().getIndex();
+        int sourceTL0Idx = sourceFrameDesc.getRTPEncoding().getIndex();
+        if (sourceTL0Idx > -1)
+        {
+            sourceTL0Idx
+                = sourceEncodings[sourceTL0Idx].getBaseLayer().getIndex();
+        }
 
+        boolean sourceTL0IsActive = sourceEncodings[sourceTL0Idx].isActive();
         if (!sourceFrameDesc.isIndependent()
-            || !sourceFrameDesc.getRTPEncoding().isActive()
-            || sourceIdx == transformState.currentIdx)
+            || !sourceTL0IsActive
+            || sourceTL0Idx == currentTL0Idx)
         {
             // An intra-codec switch requires a key frame.
-            boolean accept = sourceSSRC == transformState.currentSSRC;
 
-            if (accept)
+            long sourceSSRC = sourceFrameDesc.getRTPEncoding().getPrimarySSRC();
+            boolean accept = sourceSSRC == bitstreamController.getTL0SSRC();
+
+            if (!accept)
             {
-                onAccept(buf, off, len);
+                return false;
             }
 
-            return accept;
+            // Give the bitstream filter a chance to drop the packet.
+            return bitstreamController.accept(sourceFrameDesc, buf, off, len);
         }
 
```

Comment:
This comment might have ended up in the wrong place. I am referring to the "if" statements in SimulcastController#accept